core/src/Kokkos_MathematicalFunctions.hpp

```diff
@@ -513,6 +513,34 @@ KOKKOS_IMPL_MATH_UNARY_PREDICATE(signbit)
 #undef KOKKOS_IMPL_MATH_BINARY_FUNCTION
 #undef KOKKOS_IMPL_MATH_TERNARY_FUNCTION
 
+// non-standard math functions provided by CUDA/HIP/SYCL
+KOKKOS_INLINE_FUNCTION float rsqrt(float val) {
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+  return rsqrtf(val);
+#elif defined(KOKKOS_ENABLE_SYCL)
+  return sycl::rsqrt(val);
+#else
+  return 1.0f / Kokkos::sqrt(val);
+#endif
+}
+KOKKOS_INLINE_FUNCTION double rsqrt(double val) {
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+  return rsqrt(val);
+#elif defined(KOKKOS_ENABLE_SYCL)
+  return sycl::rsqrt(val);
+#else
+  return 1.0 / Kokkos::sqrt(val);
+#endif
+}
+inline long double rsqrt(long double val) { return 1l / Kokkos::sqrt(val); }
```

Comment:
Same comment `1l` is `long int` not `long double`.  You probably meant `1.l` (or `1.0l` for consistency with your using `1.0f` and `1.0` in the previous overloads)